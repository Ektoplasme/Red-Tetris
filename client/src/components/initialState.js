import tab from './pieces.1.json'
import pieces from './pieces.json'

export const initialTetriState = (count) => {
	const name = pieces.pieces[count]
	const tetri = {
		maxWidth: tab[name].position[0].maxWidth,
		maxHeight: tab[name].position[0].maxHeight,
		h: tab[name].position[0].height,
		type: tab[name].type,
		name: name,
		width: tab[name].position[0].width,
		x: 3,
		y: 0,
		leftSpace: tab[name].position[0].leftSpace,
		rightSpace: tab[name].position[0].rightSpace,
		downSpace: tab[name].position[0].downSpace,
		rot: 0,
		form: tab[name].position[0].form,
		color: tab[name].color,
	}
	return(tetri)
}

export const initialBoardState = () => ({
	tab: [
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0]
	],
	tetriList: []
})