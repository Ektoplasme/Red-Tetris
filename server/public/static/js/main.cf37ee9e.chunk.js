(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1:function(e){e.exports={I:{color:"red",position:[{rightSpace:0,leftSpace:0,downSpace:2,maxHeight:2,maxWidth:4,width:[0,4,0,0],height:[2,2,2,2],form:[[0,0,0,0],[1,1,1,1],[0,0,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:1,downSpace:0,maxHeight:4,width:[2,2,2,2],maxWidth:2,height:[0,4,0,0],form:[[0,1,0,0],[0,1,0,0],[0,1,0,0],[0,1,0,0]]},{rightSpace:0,leftSpace:0,maxWidth:4,downSpace:1,width:[0,0,4,0],height:[3,3,3,3],maxHeight:3,form:[[0,0,0,0],[0,0,0,0],[1,1,1,1],[0,0,0,0]]},{rightSpace:1,leftSpace:2,maxHeight:4,downSpace:0,width:[3,3,3,3],maxWidth:3,height:[0,0,4,0],form:[[0,0,1,0],[0,0,1,0],[0,0,1,0],[0,0,1,0]]}]},O:{color:"blue",position:[{rightSpace:1,maxHeight:2,leftSpace:1,width:[3,3,0,0],maxWidth:3,downSpace:2,height:[0,2,2,0],form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:1,downSpace:2,width:[3,3,0,0],maxHeight:2,height:[0,2,2,0],form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{height:[0,2,2,0],rightSpace:1,leftSpace:1,width:[3,3,0,0],maxWidth:3,maxHeight:2,downSpace:2,form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{height:[0,2,2,0],width:[3,3,0,0],maxWidth:3,rightSpace:1,leftSpace:1,maxHeight:2,downSpace:2,form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]}]},T:{color:"yellow",position:[{rightSpace:1,leftSpace:0,maxWidth:3,height:[2,2,2,0],width:[2,3,0,0],maxHeight:2,downSpace:2,form:[[0,3,0,0],[3,3,3,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,height:[0,3,2,0],width:[2,3,2,0],maxWidth:3,downSpace:1,maxHeight:3,form:[[0,3,0,0],[0,3,3,0],[0,3,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,height:[2,3,2,0],maxWidth:3,maxHeight:3,width:[0,3,2,0],downSpace:1,form:[[0,0,0,0],[3,3,3,0],[0,3,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,width:[2,3,2,0],maxWidth:3,height:[2,3,0,0],maxHeight:3,downSpace:1,form:[[0,3,0,0],[3,3,0,0],[0,3,0,0],[0,0,0,0]]}]},L:{color:"purple",position:[{rightSpace:1,leftSpace:0,maxHeight:2,width:[3,3,0,0],maxWidth:3,height:[2,2,2,0],downSpace:2,form:[[0,0,4,0],[4,4,4,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,height:[0,3,3,0],maxHeight:3,maxWidth:3,width:[2,2,3,0],downSpace:1,form:[[0,4,0,0],[0,4,0,0],[0,4,4,0],[0,0,0,0]]},{rightSpace:1,maxHeight:3,leftSpace:0,downSpace:1,maxWidth:3,width:[0,3,1,0],height:[3,2,2,0],form:[[0,0,0,0],[4,4,4,0],[4,0,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxHeight:3,maxWidth:2,width:[2,2,2,0],height:[1,3,0,0],downSpace:1,form:[[4,4,0,0],[0,4,0,0],[0,4,0,0],[0,0,0,0]]}]},J:{color:"salmon",position:[{rightSpace:1,leftSpace:0,height:[2,2,2,0],maxHeight:2,width:[1,3,0,0],maxWidth:3,downSpace:2,form:[[5,0,0,0],[5,5,5,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,maxHeight:3,maxWidth:3,height:[0,3,1,0],width:[3,2,2,0],downSpace:1,form:[[0,5,5,0],[0,5,0,0],[0,5,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,width:[0,3,3,0],height:[2,2,3,0],maxWidth:3,maxHeight:3,downSpace:1,form:[[0,0,0,0],[5,5,5,0],[0,0,5,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,maxHeight:3,width:[2,2,2,0],height:[3,3,0,0],downSpace:1,form:[[0,5,0,0],[0,5,0,0],[5,5,0,0],[0,0,0,0]]}]},Z:{color:"cyan",position:[{rightSpace:1,leftSpace:0,height:[1,2,2,0],maxHeight:2,width:[2,3,0,0],maxWidth:3,downSpace:2,form:[[6,6,0,0],[0,6,6,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:1,width:[3,3,2,0],maxHeight:3,height:[0,3,2,0],downSpace:1,form:[[0,0,6,0],[0,6,6,0],[0,6,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:0,maxHeight:3,width:[0,2,3,0],height:[2,3,3,0],downSpace:1,form:[[0,0,0,0],[6,6,0,0],[0,6,6,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,maxHeight:3,width:[2,2,1,0],height:[3,2,0,0],downSpace:1,form:[[0,6,0,0],[6,6,0,0],[6,0,0,0],[0,0,0,0]]}]},S:{color:"green",position:[{rightSpace:1,leftSpace:0,height:[2,2,1,0],width:[3,2,0,0],maxHeight:2,maxWidth:3,downSpace:2,form:[[0,7,7,0],[7,7,0,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,maxHeight:3,height:[0,2,3,0],width:[2,3,3,0],maxWidth:3,downSpace:1,form:[[0,7,0,0],[0,7,7,0],[0,0,7,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,maxWidth:3,maxHeight:3,width:[0,3,2,0],height:[3,3,2,0],downSpace:1,form:[[0,0,0,0],[0,7,7,0],[7,7,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,width:[1,1,2,0],maxHeight:3,height:[2,3,0,0],downSpace:1,form:[[7,0,0,0],[7,7,0,0],[0,7,0,0],[0,0,0,0]]}]}}},13:function(e,t,a){e.exports=a(25)},18:function(e,t,a){},19:function(e,t,a){},20:function(e,t,a){},25:function(e,t,a){"use strict";a.r(t);var i=a(0),n=a.n(i),o=a(8),r=a.n(o),c=(a(18),a(19),a(20),a(2)),h=a(9),p=a(4),l=a(1),m=a(34);function d(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter(function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable})),a.push.apply(a,i)}return a}function f(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?d(a,!0).forEach(function(t){Object(h.a)(e,t,a[t])}):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):d(a).forEach(function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))})}return e}var s={backgroundColor:"pink",width:"".concat(40,"px"),height:"".concat(40,"px"),border:"1px solid pink"},S=function(e){var t=e.empty,a=e.color,i=e.transparent;return n.a.createElement("div",{style:t?f({},s):f({},s,i?{border:"1px solid transparent",backgroundColor:"transparent"}:{border:"1px solid white",backgroundColor:"".concat(a)})})},u=Object(m.a)(function(){return{gameOverContainer:{height:"500px",width:"500px",backgroundColor:"pink"},scoreLabel:{fontSize:"40px",fontWeight:"bold",color:"pink",marginTop:"20px"},restartLabel:{fontSize:"50px",fontWeight:"bold",color:"white",padding:"10px","&:hover":{color:"salmon"}},restartButton:{cursor:"pointer","& :active":{transform:"scale(0.9, 0.9)"}},blockStyle:{backgroundColor:"pink",width:"".concat(40,"px"),height:"".concat(40,"px"),border:"1px solid pink"}}})(function(e){var t,a=function(e){var t=p.pieces[e],a={maxWidth:l[t].position[0].maxWidth,maxHeight:l[t].position[0].maxHeight,h:l[t].position[0].height,type:l[t].type,name:t,width:l[t].position[0].width,x:3,y:0,leftSpace:l[t].position[0].leftSpace,rightSpace:l[t].position[0].rightSpace,downSpace:l[t].position[0].downSpace,rot:0,form:l[t].position[0].form,color:l[t].color};return!!o(a)&&a},o=function(e){var t=W.tab,a=!0;return e.form.map(function(i,n){i.map(function(i,o){i>0&&(!t[e.y+n]||t[e.y+n][e.x+o]>0||e.x+o>9+e.rightSpace||e.x+o<0-e.leftSpace||"undefined"===typeof t[e.y+n][e.x+o])&&(a=!1)})}),a},r=e.classes,h=Object(i.useState)(0),m=Object(c.a)(h,2),d=m[0],s=m[1],u=Object(i.useState)(!1),g=Object(c.a)(u,2),x=g[0],w=g[1],y=Object(i.useState)(0),O=Object(c.a)(y,2),b=O[0],v=O[1],E=Object(i.useState)({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]}),L=Object(c.a)(E,2),W=L[0],k=L[1],I=Object(i.useState)(a(0)),j=Object(c.a)(I,2),J=j[0],T=j[1],H=Object(i.useState)(0),Z=Object(c.a)(H,2),C=Z[0],N=(Z[1],Object(i.useState)(0)),P=Object(c.a)(N,4),z=P[0],R=P[1],B=P[2],D=(P[3],Object(i.useState)(!0)),A=Object(c.a)(D,2),$=A[0],q=A[1],F=Object(i.useState)(l[p.pieces[1]]),G=Object(c.a)(F,2),K=G[0],M=G[1],Q=Object(i.useRef)(!1);Object(i.useEffect)(function(){JSON.stringify(W)===JSON.stringify({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]})&&d>0&&(s(0),v(0),a(0),T(a(0)),M(l[p.pieces[1]]),w(!1))},[W,d]);var U=function(e){(!t||38!=t.keyCode&&32!=t.keyCode&&40!=t.keyCode)&&(t=e,39===e.keyCode&&$&&T(function(e){return o(f({},e,{x:e.x+1}))?f({},e,{x:e.x+1}):e}),37===e.keyCode&&$&&T(function(e){return o(f({},e,{x:e.x-1}))?f({},e,{x:e.x-1}):e}),32===e.keyCode&&$&&(q(!1),X("hard drop",z-J.y),clearInterval(Q.current),Q.current=V(15)),38===e.keyCode&&$&&T(function(e){var t=e.rot>2?0:e.rot+1,a=f({},e,{form:l[e.name].position[t].form,rot:t,h:l[e.name].position[t].height,leftSpace:l[e.name].position[t].leftSpace,rightSpace:l[e.name].position[t].rightSpace,downSpace:l[e.name].position[t].downSpace,maxHeight:l[e.name].position[t].maxHeight,maxWidth:l[e.name].position[t].maxWidth,width:l[e.name].position[t].width,height:l[e.name].position[t].height});return o(a)?a:e}),40===e.keyCode&&(clearInterval(Q.current),Q.current=V(30)))},V=function(e){return setInterval(function(){T(function(e){if(o(f({},e,{y:e.y+1})))return f({},e,{y:e.y+1});R(e.y+1),q(!0),t=null,clearInterval(Q.current),k(function(t){return e.form.map(function(a,i){a.map(function(a,n){e.form[i][n]>0&&(t.tab[e.y+i][e.x+n]=e.form[i][n])})}),t});var i=a(d+1);return s(function(e){return M(l[p.pieces[e+2]]),e+1}),!1!==i&&o(i)?i:(w(!0),e)})},e)},X=function(e,t){var a=0;switch(e){case"row":a=[40,100,300,1200][t+1];case"hard drop":a=t+1;case"soft drop":a=t}v(function(e){return e+a})};Object(i.useEffect)(function(){return Q.current=V(1e3),x&&clearInterval(Q.current),function(){clearInterval(Q.current)}},[d,x]),Object(i.useEffect)(function(){return document.addEventListener("keydown",U),document.addEventListener("keyup",function(e){40===e.keyCode&&(clearInterval(Q.current),Q.current=V(1e3)),32!==t&&(t=null)}),x&&document.removeEventListener("keydown",U),function(){document.removeEventListener("keydown",U)}},[d,x,$,J,C,z,B]);var Y=["red","blue","yellow","purple","salmon","cyan","green"],_=function(){var e=W.tab,t=[];e.map(function(a,i){return a.every(function(e){return e>0})&&t.push(i),e}),t.length&&(function(e,t){var a=t;e.map(function(t){return a.splice(t,1),a.splice(0,0,[0,0,0,0,0,0,0,0,0,0]),e}),k({tetriList:W.tetriList,tab:a})}(t,e),X("row",t.length))};Object(i.useEffect)(function(){_()});var ee=J.form;return x?n.a.createElement("div",{className:"flex column center alignCenter",style:{height:"100vh"}},n.a.createElement("div",{className:"flex column center alignCenter ".concat(r.gameOverContainer)},n.a.createElement("div",{style:{fontSize:"60px",fontWeight:"bold",color:"salmon"}},"Sorry, you lose"),n.a.createElement("div",{style:{fontSize:"50px",fontWeight:"bold",color:"white"}},"SCORE: ",b),n.a.createElement("div",{className:"flex column center alignCenter ".concat(r.restartButton)},n.a.createElement("div",{className:r.restartLabel,onClick:function(){k({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]})}},"RESTART")))):n.a.createElement("div",{className:"flex"},n.a.createElement("div",{className:"absolute",style:{height:"500px",left:"100px"}},n.a.createElement("div",{className:r.scoreLabel},"SCORE: ",b)),n.a.createElement("div",{className:"absolute",style:{top:"100px",left:"100px"}},W.tab.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:Y[e-1]}):n.a.createElement(S,{empty:!0}))}))}),n.a.createElement("div",{className:"absolute",style:{top:"".concat(40*J.y+2*J.y,"px"),left:"".concat(40*J.x+2*J.x,"px")}},ee.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:J.color}):n.a.createElement(S,{transparent:!0}))}))}))),n.a.createElement("div",{className:"absolute flex center alignCenter column",style:{backgroundColor:"lightblue",left:"600px",top:"100px"}},n.a.createElement("div",{style:{fontSize:"40px",fontWeight:"bold",color:"navy",marginTop:"20px"}},"Next:"),n.a.createElement("div",{className:"relative",style:{top:"10px",padding:"10px"}},K.position[0].form.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:K.color}):n.a.createElement(S,{transparent:!0}))}))}))))});Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));r.a.render(n.a.createElement(u,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then(function(e){e.unregister()})},4:function(e){e.exports={pieces:["T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I"]}}},[[13,1,2]]]);
//# sourceMappingURL=main.cf37ee9e.chunk.js.map