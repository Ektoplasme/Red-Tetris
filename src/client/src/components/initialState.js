import tab from './pieces.1.json'
import { isEmpty } from 'lodash'

export const initialTetriState = (count, data) => {
	let name = "T";
	if (!isEmpty(data))
		name = data[count];
	const tetri = {
		maxWidth: tab[name].position[0].maxWidth,
		maxHeight: tab[name].position[0].maxHeight,
		h: tab[name].position[0].height,
		type: tab[name].type,
		name: name,
		width: tab[name].position[0].width,
		x: 3,
		y: 0,
		leftSpace: tab[name].position[0].leftSpace,
		rightSpace: tab[name].position[0].rightSpace,
		downSpace: tab[name].position[0].downSpace,
		rot: 0,
		form: tab[name].position[0].form,
		color: tab[name].color,
	}
	return(tetri)
}

export const initialBoardState = () => (
	[
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0],
		[0,0,0,0,0,0,0,0,0,0]
	])