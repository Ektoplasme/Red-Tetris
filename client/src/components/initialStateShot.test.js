import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { initialTetriState } from './initialState.js'

describe("initialState SnapShotTest", () => {
	test("Component Match initialTetrisState Snapshot", () => {
		const blockCmp = renderer.create(< initialTetriState />).toJSON();	
		expect(blockCmp).toMatchSnapshot();
	})
});
