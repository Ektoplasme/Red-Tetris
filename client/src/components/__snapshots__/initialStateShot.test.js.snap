// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState SnapShotTest Component Match initialTetrisState Snapshot 1`] = `<initialTetriState />`;
